### PR TITLE
Remove Docker

### PR DESCRIPTION
**Pull request by hunchr**
TICKET-21134

- [x] Remove Docker and Make
- [x] Fix failing specs and linters (apparently the linters weren't running correctly before on Docker)
- [x] Update README (new setup guide)
- [x] Fix chrome webdriver bug (works just by removing Docker)

This will make the setup process way easier

### CODE REVIEW

**Review by schmijos — DISMISSED** [review]
Please remove the controller changes from this PR. Afterwards GTM.

File: app/controllers/time_tracker_controller.rb
[the remark this answers is not in the document]
Comment (hunchr) [reply app/controllers/time_tracker_controller.rb:1]: I wanted to remove the rubocop:disable

**Comment by hunchr** [comment]
I reverted the changes to fix the linters. I will do this in another pull request instead.

**Review by sihu — COMMENTED** [review]
i tried the setup and had two issues:

1. ruby version: i'd like to not use asdf-default version but the .ruby-version file
2. i did not have a running mysql. this we can easily fix with a prequisities (or i gave you some ideas)
3. can we also delete `shell.nix`? is there still a use for it?


After that we are good to merge.

File: Rakefile [hunk 1]
```diff
@@ -116,25 +88,21 @@ end
...
-
-  sh "docker compose exec -e RAILS_ENV=test redmine rake #{command}"
+  file = "test/#{file}" if file
+  sh "cd ../.. && RAILS_ENV=test rake test TEST=plugins/redmine_tracky/#{file}"
 end
 
 desc 'Test JavaScript code.'
 task :test_javascript do
-  sh 'docker compose exec -w /app/assets.src node npm run test'
+  # TODO: sh 'cd assets.src && npm run test'
```
[the remark this answers is not in the document]
Comment (hunchr) [reply Rakefile:97]: we don't have js tests so this might be a todo for another pr

**Comment by hunchr** [comment]
> 1. ruby version: i'd like to not use asdf-default version but the .ruby-version file

fixed
> 2. i did not have a running mysql. this we can easily fix with a prequisities (or i gave you some ideas)

added prerequisites and note to adjust the database file in readme
> 3. can we also delete `shell.nix`? is there still a use for it?

I wouldn't risk it because nixOS users need it, I guess

File: Rakefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (sihu) [reply Rakefile:98]: then let's delete the command for now. TODOS will remain in a project.

File: bin/setup [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+
+# Configure Database
+db=plugins/redmine_tracky/.github/database.yml
+[[ -z "$GITHUB_ACTIONS" ]] && db=config/database.yml.example
+cp $db config/database.yml
+
+bundle install -j 4
+[[ -z "$GITHUB_ACTIONS" ]] && rake db:drop
+rake db:create db:migrate db:seed
+rake redmine:plugins:redmine_tracky:install
```
Comment (sihu) [inline bin/setup:19]:
```suggestion
rake redmine:plugins:redmine_tracky:install
rake redmine:sample_data:seed
```

redmine/plugins/redmine_tracky/lib/tasks/seed.rake
```rb
# frozen_string_literal: true

namespace :redmine do
  namespace :sample_data do
    desc 'Seeding sample values'
    task :seed => :environment do
      user_amount = 20
      project_amount = 10
      issue_per_project_amount = 10
      members_per_project = 10
      timelog_per_issue = 2
      activity = TimeEntryActivity.create!(name: 'Entwicklung')
      timelog_start = Date.new(2010, 1, 1).to_time
      timelog_end = Time.zone.today.to_time

      IssueStatus.create!(name: 'CURRENT')

      Tracker.create!(name: "NOW", default_status: IssueStatus.first)
      IssuePriority.create!(name: 'EOD')

      print 'Create projects'
      project_amount.times do |project_num|
        project = Project.create!(
          name: "Project #{project_num}",
          identifier: "project-#{project_num}",
          is_public: false,
          description: "Description for project #{project_num}"
        )

        # Assign members
        issue_per_project_amount.times do |issue_num|
          Issue.create!(
            subject: "Issue #{issue_num} for project #{project_num}",
            description: "Description for Issue #{issue_num} in project #{project_num}",
            tracker: Tracker.first,
            author: User.first,
            status: IssueStatus.first,
            priority: IssuePriority.first,
            project_id: project.id
          )
        end
        print '.'
      end
    end
  end
end
```

**Review by sihu — APPROVED** [review]
Nice! looking forward to not use docker anymore 🎉  thx for your work!

**Comment by hunchr** [comment]
why did you just recommit everything?

**Pull request merged**